specification/search/data-plane/Azure.Search/readme.md

```diff
@@ -170,32 +155,22 @@ directive:
           .replace(/(public IndexAction setAdditionalProperties\(Map<String, Object> additionalProperties\) {\s+this.additionalProperties = additionalProperties;\s+return this;\s+})/g, "@SuppressWarnings(\"unchecked\")\n    public IndexAction<T> setDocument(T document) {\n        if (document instanceof Map) {\n            this.properties = (Map<String, Object>) document;\n            this.document = null;\n        } else {\n            this.document = document;\n            this.properties = null;\n        }\n        return this;\n    }")
           .replace(/(public IndexAction setActionType\(IndexActionType actionType\) {)/g, "public IndexAction<T> setActionType(IndexActionType actionType) {")
 
-    # Enable configuration of RestProxy serializer
-    - from: SearchIndexRestClientBuilder.java
+    # Enable IndexBatchBase to be used as a generic type
+    - from: IndexBatchBase.java
       where: $
       transform: >-
           return $
-          .replace(/(package com.azure.search.implementation;)/g, "$1\nimport com.azure.core.util.serializer.SerializerAdapter;")
-          .replace(/(\* The HTTP pipeline to send requests through)/g, "\* The serializer to use for requests\n     \*\/\n    private SerializerAdapter serializer;\n\n    \/\*\*\n     \* Sets The serializer to use for requests.\n     \*\n     \* @param serializer the serializer value.\n     \* @return the SearchIndexRestClientBuilder.\n     \*\/\n    public SearchIndexRestClientBuilder serializer\(SerializerAdapter serializer\) {\n        this.serializer = serializer;\n        return this;\n    }\n\n    \/\*\n     $1")
-          .replace(/(new SearchIndexRestClientImpl\(pipeline)/g, "$1, serializer")
-          .replace(/(this.pipeline = RestProxy.createDefaultPipeline\(\);\s+})/g, "$1\n        if \(serializer == null\) {\n            this.serializer = JacksonAdapter.createDefaultSerializerAdapter\(\);\n        }")
-
-    # Enable IndexBatchImpl to be used as a generic type
-    - from: IndexBatchImpl.java
-      where: $
-      transform: >-
-          return $
-          .replace(/(public final class IndexBatchImpl)/g, "public class IndexBatchImpl<T>")
+          .replace(/(public final class IndexBatchBase)/g, "public class IndexBatchBase<T>")
```

Comment:
This can be made abstract

```suggestion
          .replace(/(public final class IndexBatchBase)/g, "public abstract class IndexBatchBase<T>")
```

Reply:
Could we also change the other base classes to be abstract in this change? @brjohnstmsft